### DISCUSSION TITLE
Expose router.find from FindMyWay

### DISCUSSION THREAD

**Discussion by rizrmd** [fastify/fastify#3686] [Ideas] [answered]
Hi.

I am creating proxy that intercepts based on url. For that I needed to check whether some request already has routes for its url. When url has some routes defined for it, then call the routes instead of passing it to proxy.

I don't think it's currently possible to do this efficiently in fastify because router.find from FindMyWay is not accessible from outside.
Or maybe even better, expose FindMyWay router itself as `rawRouter` ?

**Comment by Eomm** [comment, marked as answer]
Yes, the router is not exposed to run the `find` method, only the `lookup`.

I don't have a clear picture of your use case, but I think it is fine to expose the `router.find` function

**Reply by rizrmd** [reply]
If find method is exposed, one can build dynamic router in fastify by catching all request, testing the url, reusing defined routes when desired.

In my case, when routes is not available for particular url, pass them to proxy.
